File: finger/finger0/datos.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#define MAX 10000

typedef struct datos {
	size_t positivos, cantidad;
	unsigned long longitud;
	bool amazing_positivo;	
} datos_t;


void agregar_informacion(datos_t *datos, char *linea){
	int i;
	for (i = 0; i < strlen(linea) && linea[i] != '\t'; i++){		
	}
	i++;
	if (linea[i] == '1'){
		datos->positivos++;
	}
	datos->cantidad++;

}


void algoritmo_1(){
	FILE *archivo = fopen("labeledTrainData.tsv", "r");
	int caracter;
	char buffer[MAX];
	size_t indice = 0;
	datos_t *datos = malloc(sizeof(datos_t));
	datos->cantidad = 0;
	datos->positivos = 0;
	size_t cantidad = 0;
	while ((caracter = fgetc(archivo)) != EOF){		
		if (caracter == '\n' && caracter != EOF){
			cantidad++;
			buffer[indice++] = '\0';
			/*if (cantidad >= 24500)
				printf("%s\n", buffer);*/
			indice = 0;
			agregar_informacion(datos, buffer);
		} else if (caracter != EOF){
			buffer[indice++] = caracter;
		}		
	}
	fclose(archivo);
	printf("Cantidad: %zd\n", cantidad);
	printf("Cantidad positivos: %zd\n", datos->positivos);
	printf("Cantidad negativos: %zd\n", datos->cantidad - datos->positivos);
	free(datos);
}



int main(void){
	algoritmo_1();
}

File: finger/finger0/datos.py
#!/usr/python
import csv
import sys
from time import *
"""
Finalidad de este script
Con el archivo "labeledTrainData.tsv" obtener las siguientes estadisticas

1) Reviews positivos y negativos: Cantidad
positivos + negativos = 25000
P = 12500
N = 12500

2) Longitud del comentario del review (promedio)
1326


3) Verificar si encuentro la palabra "amazing" y review = 0
Amazing no se usa siempre en reviews positivos

"""
def algoritmo_2(file):
	positivos = 0
	negativos = 0
	longitud = 0
	promedio = 0.0
	max = 0
	min = 100
	amazing_positivo = True
	primera_linea = True
	comienzo = time()	
	with open(file) as tsv:
		for line in csv.reader(tsv, dialect="excel-tab"):
			if (primera_linea):
				primera_linea = False
				continue
			if (line[1] == '1'):
				positivos += 1
			elif (line[1] == '0'):
				negativos += 1
			if (len(line[2]) > max):
				max = len(line[2])
			if (len(line[2]) < min):
				min = len(line[2])
			promedio += (len(line[2]) - 2)
			if (("amazing" in line[2].lower()) and (line[1] == '0')):				
				amazing_positivo = False
		
	comienzo = time() - comienzo
	print "Cantidad positivos: " + str(positivos)
	print "Cantidad negativos: " + str(negativos)
	print "Promedio longitud comentarios: " + str(promedio / (positivos + negativos))
	print "Maximo longitud: " + str(max)
	print "Minimo longitud: " + str(min)
	print "Amazing comentario positivo: " + str(amazing_positivo)
	print "Tardo: " + str(comienzo) + " segundos"


def algoritmo_1(file):
	cantidad = 0
	positivos = 0
	longitud = 0.0
	amazing_positivo = True
	comienzo = time()
	primera_linea = True
	with open(file) as tsv:
		for line in csv.reader(tsv, dialect="excel-tab"):
			if (primera_linea):
				primera_linea = False
				continue
			cantidad += 1
			positivos += int(line[1])
			longitud += len(line[2])						
			if ("amazing" in line[2].lower() and (line[1] == '0')):
				amazing_positivo = False
	comienzo = time() - comienzo
	print "Cantidad positivos: " + str(positivos)
	print "Cantidad negativos: " + str(cantidad - positivos)
	print "Promedio longitud comentarios: " + str(longitud/cantidad)
	print "Amazing comentario positivo: " + str(amazing_positivo)
	print "Tardo: " + str(comienzo) + " segundos"


def main():
	if (len(sys.argv) > 1):
		file = sys.argv[1] 
	else:
		file = "labeledTrainData.tsv"
	algoritmo_2(file)

main()

